File: 정수빈/LeetCode/medium/count-and-say.py
# https://leetcode.com/problems/count-and-say/

class Solution:
    def countAndSay(self, n: int) -> str:
        
        def recurive(num):
            if num == 1:
                return "1"
            
            prev = recurive(int(num)-1)
            ans = ""
            order = ""
            prev_c = prev[0]
            count = 1
            for i in range(1, len(prev)):
                if prev_c == prev[i]:
                    count += 1
                else:
                    ans += str(count) + prev_c
                    prev_c = prev[i]
                    count = 1
            ans += str(count) + prev_c
            return ans
        return recurive(n)
            

File: 장세훈/LeetCode/medium/CountAndSay.java
class CountAndSay {
    /*
        https://leetcode.com/problems/count-and-say/
     */

    public String countAndSay(int n) {
        String say = "";

        for (int i = 1; i <= n; i++) {
            if (i == 1) {
                say = "1";
            } else {
                String base = say.charAt(0) + "";
                int count = 1;
                StringBuilder newSay = new StringBuilder();
                for (int j = 1; j < say.length(); j++) {
                    if (base.equals(say.charAt(j) + ""))
                        count++;
                    else {
                        newSay.append(count).append(base);
                        count = 1;
                        base = say.charAt(j) + "";
                    }
                }
                newSay.append(count).append(base);
                say = newSay.toString();
            }

        }

        return say;
    }
}